srcpkgs/mercury/template

```diff
@@ -0,0 +1,150 @@
+distfiles="http://dl.mercurylang.org/release/mercury-srcdist-${version}.tar.xz"
+checksum=4ab0734aaa66808c91ffe7f54c7d818bdc4f0247aa255932e1e3d50610db4150
+nocross="Mercury cannot be cross-compiled to different architectures"
+noverifyrdeps=yes
+noshlibprovides=yes
+shlib_requires="libc.so.6"
+
+# asm_fast is preferable, but does not work on all architectures.
+case "$XBPS_MACHINE" in
+	*x86_64*|*arm*)
+		_LLDS_GRADE=asm_fast ;;
+	*i686*)
+		_LLDS_GRADE=reg ;;
+	*)
+		_LLDS_GRADE=none ;;
+esac
+
+_LIBGRADES="hlc.gc,hlc.gc.ssdebug,hlc.gc.memprof,${_LLDS_GRADE}.gc,${_LLDS_GRADE}.gc.stseg,${_LLDS_GRADE}.gc.decldebug.stseg,${_LLDS_GRADE}.gc.profdeep.stseg"
+
+# The parallel grades have some issues on musl systems.
+if [ "$XBPS_LIBC" != musl ] ; then
+	_LIBGRADES="${_LIBGRADES},hlc.par.gc,${_LLDS_GRADE}.par.gc.stseg"
+fi
+
+do_configure() {
+	# Prefer the low-level C grade as default, unless it is none.
+	if [ "${_LLDS_GRADE}" != "none" ] ; then
+		DEFAULT_GRADES="${_LLDS_GRADE}.gc.stseg"
+	else
+		DEFAULT_GRADES=hlc.gc
+	fi
+
+	./configure --prefix=/usr/ \
+		--enable-new-mercuryfile-struct \
+		--enable-gc-mmap --enable-gc-munmap \
+		--without-readline --with-editline \
+		--enable-libgrades="${_LIBGRADES}" --with-llds-base-grade="${_LLDS_GRADE}" --with-default-grade="${DEFAULT_GRADES}"
+}
+
+do_build() {
+	make PARALLEL="${makejobs}" DESTDIR=$(pwd)
+	make PARALLEL="${makejobs}" DESTDIR=$(pwd) install
+}
+
+mercury-libs_package() {
+	license="custom:LGPL-2.0-only-linking-exception"
+	short_desc+=" - runtime libraries"
+	noverifyrdeps=yes
+	noshlibprovides=yes
+	pkg_install() {
+		vmkdir /usr/lib/mercury/lib
+		vcopy usr/lib/mercury/lib/libgc*.so  /usr/lib/mercury/lib
+		vlicense COPYING.LIB
+		for GRADE in $(echo $_LIBGRADES | sed 's/,/ /g') ; do
+			# Ignore profiling and debug grades for the runtime libraries.
+			# Also skip pregen, we don't need that after bootstrapping.
+			case $GRADE in
+				*prof*|*debug*|hlc.gc.pregen)
+					;;
+				*)
+					vmkdir /usr/lib/mercury/lib/$GRADE
+					vcopy usr/lib/mercury/lib/$GRADE/lib*.so /usr/lib/mercury/lib/$GRADE
+					;;
+			esac
+		done
+	}
+}
+
+mercury-tools_package() {
+	depends="mercury-libs>=${version}_${revision}"
+	short_desc+=" - extra tools"
+	noverifyrdeps=yes
+	noshlibprovides=yes
+	pkg_install() {
+		vman usr/share/man/man1/mprof.1
+		vman usr/share/man/man1/mprof_merge_runs.1
+		vlicense COPYING
+		vbin usr/bin/mcov
+		vbin usr/bin/mdprof
+		vbin usr/bin/mercury_profile
+		vbin usr/bin/mprof
+		vbin usr/bin/mprof_merge_runs
+		vbin usr/bin/mtc
+		vbin usr/bin/mtc_diff
+		vbin usr/bin/mtc_union
+	}
+}
+
+do_install() {
+	vmkdir /usr/lib/mercury/lib
+	vmkdir /usr/lib/mercury/conf
+	vman usr/share/man/man1/c2init.1
+	vman usr/share/man/man1/mdb.1
+	vman usr/share/man/man1/mercury_config.1
+	vman usr/share/man/man1/mgnuc.1
+	vman usr/share/man/man1/ml.1
+	vman usr/share/man/man1/mmake.1
+	vman usr/share/man/man1/mmc.1
+	vman usr/share/man/man1/mtags.1
+	vlicense COPYING
```

Comment:
should not be needed if this is GPL

Reply:
If this can't be set on the subpackage, should I set the entire license to be `GPL, custom:LGPL-2.0-only-linking-exception` then?

For reference, the runtime libraries are LGPL with a static linking exception, and the compiler toolchain itself is GPL.

Reply:
yeah `GPL-2.0-only, custom:LGPL-2.0-only-linking-exception` and only install the custom one in the packages

Reply:
Done